geomesa-blobstore/geomesa-blobstore-api/src/main/scala/org/locationtech/geomesa/blob/api/BlobstoreServlet.scala

```diff
@@ -8,43 +8,53 @@
 
 package org.locationtech.geomesa.blob.api
 
-import com.typesafe.scalalogging.LazyLogging
+import java.io.{IOException, File}
+import java.nio.file.Files
+import java.nio.file.attribute.PosixFilePermission._
+import java.nio.file.attribute.PosixFilePermissions
+
 import org.locationtech.geomesa.accumulo.data.{AccumuloDataStore, AccumuloDataStoreFactory}
 import org.locationtech.geomesa.blob.core.AccumuloBlobStore
 import org.locationtech.geomesa.web.core.GeoMesaScalatraServlet
-import org.scalatra.{NotFound, Ok}
+import org.scalatra._
+import org.scalatra.servlet.{FileUploadSupport, MultipartConfig, SizeConstraintExceededException}
 
 import scala.collection.JavaConversions._
+import scala.util.control.NonFatal
 
-class BlobstoreServlet extends GeoMesaScalatraServlet with LazyLogging {
+class BlobstoreServlet extends GeoMesaScalatraServlet with FileUploadSupport with GZipSupport {
   override def root: String = "blob"
 
-  var abs: AccumuloBlobStore = null
-
-  // TODO: Revisit configuration and persistence of configuration.
-  // https://geomesa.atlassian.net/browse/GEOMESA-958
-  // https://geomesa.atlassian.net/browse/GEOMESA-984
-  post("/ds") {
+  // caps blob size at 10MB
+  configureMultipartHandling(
+    MultipartConfig(
+      maxFileSize = Some(50*1024*1024),
```

Comment:
ideally this would be configurable in the UI for geoserver...or at least a system prop to override it